File: fries-leetcode-training/16-input-output/01-basic-input-output.cpp
/*
01-基础输入输出处理

题目描述：
演示基本的输入输出处理，包括整数、字符串、数组的读取和输出。

输入格式：
第一行：整数n
第二行：n个整数，空格分隔
第三行：字符串

输出格式：
第一行：整数n
第二行：n个整数，空格分隔
第三行：字符串

示例：
输入：
5
1 2 3 4 5
hello world

输出：
5
1 2 3 4 5
hello world
*/

#include <iostream>
#include <vector>
#include <string>
using namespace std;

void basicInputOutput() {
    int n;
    cin >> n;
    
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    
    // 读取字符串（跳过换行符）
    cin.ignore();
    string s;
    getline(cin, s);
    
    // 输出
    cout << n << endl;
    for (int i = 0; i < n; i++) {
        cout << arr[i];
        if (i < n - 1) cout << " ";
    }
    cout << endl;
    cout << s << endl;
}

void testCases() {
    cout << "=== 基础输入输出测试 ===" << endl;
    
    // 模拟输入数据
    int n = 5;
    vector<int> arr = {1, 2, 3, 4, 5};
    string s = "hello world";
    
    cout << "整数: " << n << endl;
    cout << "数组: ";
    for (int i = 0; i < arr.size(); i++) {
        cout << arr[i];
        if (i < arr.size() - 1) cout << " ";
    }
    cout << endl;
    cout << "字符串: " << s << endl;
    
    cout << "\n格式化输出:" << endl;
    cout << n << endl;
    for (int i = 0; i < arr.size(); i++) {
        cout << arr[i];
        if (i < arr.size() - 1) cout << " ";
    }
    cout << endl;
    cout << s << endl;
}

int main() {
    // 运行测试
    testCases();
    
    // 交互式输入（取消注释以启用）
    // basicInputOutput();
    
    return 0;
}

File: fries-leetcode-training/16-input-output/01-basic-input-output.py
"""
01-基础输入输出处理

题目描述：
演示基本的输入输出处理，包括整数、字符串、数组的读取和输出。

输入格式：
第一行：整数n
第二行：n个整数，空格分隔
第三行：字符串

输出格式：
第一行：整数n
第二行：n个整数，空格分隔
第三行：字符串

示例：
输入：
5
1 2 3 4 5
hello world

输出：
5
1 2 3 4 5
hello world
"""

def basic_input_output():
    """基础输入输出处理"""
    # 读取整数
    n = int(input())
    
    # 读取整数数组
    arr = list(map(int, input().split()))
    
    # 读取字符串
    s = input().strip()
    
    # 输出
    print(n)
    print(*arr)  # 输出数组，空格分隔
    print(s)

def test_cases():
    """测试用例"""
    print("=== 基础输入输出测试 ===")
    
    # 模拟输入
    test_input = """5
1 2 3 4 5
hello world"""
    
    # 分割输入行
    lines = test_input.strip().split('\n')
    
    # 处理输入
    n = int(lines[0])
    arr = list(map(int, lines[1].split()))
    s = lines[2]
    
    # 输出结果
    print(f"整数: {n}")
    print(f"数组: {arr}")
    print(f"字符串: {s}")
    
    print("\n格式化输出:")
    print(n)
    print(*arr)
    print(s)

if __name__ == "__main__":
    # 运行测试
    test_cases()
    
    # 交互式输入（取消注释以启用）
    # basic_input_output()
